src/app/Common/Services/TokenService.cs

```diff
@@ -57,6 +61,19 @@ public async Task<Token> FindAsync(string tokenValue)
             return await FindOneAsync(new TokenFilter { Value = tokenValue });
         }
 
+        public UserToken GetUserToken(string accessToken)
+        {
+            return (from token in _tokenRepository.GetQueryable()
+                    join user in _userRepository.GetQueryable() on token.UserId equals user.Id
+                    where token.Type == TokenTypeEnum.Access && token.Value == accessToken
+                    select new UserToken
+                    {
+                        UserId = token.UserId,
+                        ExpireAt = token.ExpireAt,
+                        UserRole = user.Role
+                    }).FirstOrDefault();
```

Comment:
Typically you won't often see queries to MongoDB in such style often. Usually there'll be two separate queries, or we can store user role in tokens and make a single query without join. Though in this case the query works as expected and results in a single `aggregation` query to DB with a `$lookup` stage.

The only thing I'm not sure about - is it executed asynchronously? `FirstOrDefault` is called on `IQueryable`. But if it was `IMongoQueryable`, we'd be able to call `FirstOrDefaultAsync` method. I'd check this and rewrite the query if necessary, or consider the options I described.